test/benchmarks/material/button-harness/protractor.perf-spec.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {HarnessLoader} from '@angular/cdk/testing';
+import {MatButtonHarness} from '@angular/material/button/testing/button-harness';
+import {ProtractorHarnessEnvironment} from '@angular/cdk/testing/protractor';
+import {$$, element, by, browser} from 'protractor';
+import {benchmark} from './protractor-benchmark-utilities';
+import {NUM_BUTTONS} from './constants';
+
+const FIRST_BUTTON = '0';
+const MIDDLE_BUTTON = `${Math.floor(NUM_BUTTONS / 2)}`;
+const LAST_BUTTON = `${NUM_BUTTONS - 1}`;
+
+describe('performance baseline for the protractor harness', () => {
+  beforeEach(async () => {
+    await browser.get('');
+  });
+
+  it('(baseline) should retrieve all of the buttons', async () => {
+    await benchmark('(baseline) get every button', async () => {
+      await $$('.mat-button');
+    });
+  });
+
+  it('(baseline) should click the first button', async () => {
+    await benchmark('(baseline) click first button', async () => {
+      await element(by.buttonText(FIRST_BUTTON)).click();
+    });
+  });
+
+  it('(baseline) should click the middle button', async () => {
+    await benchmark('(baseline) click middle button', async () => {
+      await element(by.buttonText(MIDDLE_BUTTON)).click();
+    });
+  });
+
+  it('(baseline) should click the last button', async () => {
+    await benchmark('(baseline) click last button', async () => {
+      await element(by.buttonText(LAST_BUTTON)).click();
+    });
+  });
+
+  it('(baseline) should click all of the buttons', async () => {
+    await benchmark('(baseline) click every button', async () => {
+      const buttons = $$('.mat-button');
+      await buttons.each(async (button) => await button!.click());
+    });
+  });
+});
+
+describe('performance tests for the protractor harness', () => {
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    await browser.get('');
+    loader = ProtractorHarnessEnvironment.loader();
+  });
+
+  it('should retrieve all of the buttons', async () => {
```

Comment:
Do you know if there is a chance to share the tests across TestBed and Protractor? The code itself seems very similar, except of different functions for:

* `Before Setup`
* Getting a button in the baseline (e.g. `by.buttonText` and the TestBed equivalent).
* The benchmark wrapping function

Those could be passed through arguments to a "shared" test function. Similar to how we share MDC/non-MDC harness tests. That would help us reducing duplication and increases code maintainability/consistency.

Reply:
I'm going to look into this because it is tedious to keep going back and forth making sure both tests are the same.

Reply:
After looking into this, it seems like creating a shared spec might be a bit overkill. Especially considering Miles doesn't see us adding many more component harness perf tests beyond the ones in this PR

Reply:
Yeah, works for me. It was not clear to me in what fashion we want to maintain these harness tests.